### PR TITLE
feat(chat-messages): add support for syntax highlighting, copying and downloading csv to markdown renderer

### PR DESCRIPTION
**Pull request by Killusions**
- add code copy button to markdown renderer
- add table csv download button to markdown renderer
- complete unfinished code blocks for streaming in markdown renderer
- add optional support for code syntax highlighting in markdown renderer
- add optional support for latex formula rendering in markdown renderer (commonly used by AI chatbots)

New feature showcase:

ScreenFloat Shot of Arc on 2025-12-06 at 15-57-58

ScreenFloat Shot of Arc on 2025-12-06 at 15-57-37

---

- [x] I confirm that this MR follows the [contribution guidelines](https://github.com/siemens/element/blob/main/CONTRIBUTING.md).

This PR enhances the markdown renderer used in chat messages with interactive features, rendering options, theming, examples, and supporting changes.

Highlights
- Adds copy-to-clipboard button for code blocks and per-table CSV download button (customizable, translatable labels).
- Completes streaming-safe code block handling using placeholders and per-instance caching; renderer now returns HTMLElement and maintains caches for code blocks, tables, and LaTeX.
- Optional syntax highlighting (highlight.js) and optional LaTeX rendering (KaTeX) for inline and display math; examples updated to demonstrate both.
- Exposes MarkdownRendererOptions and new SiMarkdownRendererComponent inputs: disableCopyButton, disableDownloadButton, syntaxHighlighter, latexRenderer, copyButtonLabel, downloadButtonLabel; getMarkdownRenderer now accepts an options parameter.
- Adds helpers: sanitizeHtmlWithStyles and LRU-style caches for DOM/string reuse; extensive renderer refactor and placeholder/post-render pipeline.
- Adds SCSS highlight theme and layout for code/table overlays and KaTeX styling; sample markdown and showcase images updated.
- Expands tests for LaTeX and table/code edge cases; updates Playwright snapshots and example configs.
- Adds devDependencies (highlight.js, katex, @types/katex) and includes KaTeX CSS in example build.
- Chat container: debounced scrolling during streaming and new public scrollToBottom()/scrollToTop() methods.

Backward compatibility
- Features are opt-in via new options/inputs; intended to be non-breaking for existing usage.

---

[Documentation](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/).
[Examples](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/element-examples/).
[Dashboards Demo](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/dashboards-demo/).
[Playwright report](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/playwright-report/).

**Coverage Reports:**

Code Coverage

- [charts-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/coverage/charts-ng/)
- [dashboards-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/coverage/dashboards-ng/)
- [element-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/coverage/element-ng/)
- [element-translate-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/coverage/element-translate-ng/)
- [maps-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/coverage/maps-ng/)
- [native-charts-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1158/pages/coverage/native-charts-ng/)

### CODE REVIEW

File: projects/element-theme/src/styles/components/_markdown-hljs-theme.scss [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    color: variables.$element-text-secondary;
+  }
+
+  // Function names, class names
+  .hljs-title,
+  .hljs-section,
+  .hljs-selector-id {
+    color: variables.$element-text-primary;
+    font-weight: variables.$si-font-weight-semibold;
+  }
```
[the remark this answers is not in the document]
Comment (Killusions) [reply projects/element-theme/src/styles/components/_markdown-hljs-theme.scss]: Removed.

File: src/app/examples/si-chat-messages/si-ai-message.ts [hunk 2]
```diff
@@ -17,8 +19,28 @@ import { LOG_EVENT } from '@siemens/live-preview';
...
+      if (language && hljs.getLanguage(language)) {
+        try {
+          return hljs.highlight(code, { language }).value;
+        } catch {
+          // If highlighting fails, fall back to no highlighting
+        }
+      }
+      return undefined;
+    }
+  });
```
[the remark this answers is not in the document]
Comment (Killusions) [reply src/app/examples/si-chat-messages/si-ai-message.ts]: Nope, these are demo examples

File: projects/element-ng/markdown-renderer/markdown-renderer.ts [hunk 3]
```diff
@@ -81,29 +155,142 @@ export const getMarkdownRenderer = (sanitizer: DomSanitizer): ((text: string) =>
...
+        const blob = new Blob([csv], { type: 'text/csv;charset=utf-8;' });
+        const link = document.createElement('a');
+        const url = URL.createObjectURL(blob);
+        link.setAttribute('href', url);
+        link.setAttribute('download', 'table.csv');
+        link.style.visibility = 'hidden';
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+        URL.revokeObjectURL(url);
```
[the remark this answers is not in the document]
Comment (Killusions) [reply projects/element-ng/markdown-renderer/markdown-renderer.ts]: This is fine.

File: src/app/examples/si-chat-messages/si-user-message.ts [hunk 4]
```diff
@@ -20,8 +22,28 @@ import { LOG_EVENT } from '@siemens/live-preview';
 export class SampleComponent {
   logEvent = inject(LOG_EVENT);
   private sanitizer = inject(DomSanitizer);
+  private translate = injectSiTranslateService();
 
-  protected markdownRenderer = getMarkdownRenderer(this.sanitizer);
+  protected markdownRenderer = getMarkdownRenderer(this.sanitizer, {
+    copyCodeButton: 'SI_MARKDOWN_RENDERER.COPY',
+    downloadTableButton: 'SI_MARKDOWN_RENDERER.DOWNLOAD',
```
Comment (kfenner) [inline src/app/examples/si-chat-messages/si-user-message.ts]: Don't include this in the example as the properties are defaulted.

```suggestion
```

**Review by kfenner — CHANGES_REQUESTED** [review]
just a couple minor findings

File: src/app/examples/si-chat-messages/si-ai-message.ts [hunk 5]
```diff
@@ -17,8 +19,28 @@ import { LOG_EVENT } from '@siemens/live-preview';
 export class SampleComponent {
   logEvent = inject(LOG_EVENT);
   private sanitizer = inject(DomSanitizer);
+  private translate = injectSiTranslateService();
 
-  protected markdownRenderer = getMarkdownRenderer(this.sanitizer);
+  protected markdownRenderer = getMarkdownRenderer(this.sanitizer, {
+    copyCodeButton: 'SI_MARKDOWN_RENDERER.COPY',
+    downloadTableButton: 'SI_MARKDOWN_RENDERER.DOWNLOAD',
```
[the remark this answers is not in the document]
Comment (Killusions) [reply src/app/examples/si-chat-messages/si-ai-message.ts]: They are actually not defaulted since they are only defaulted on the components, not the standalone function.

File: src/app/examples/si-chat-messages/si-chat-container.ts [hunk 6]
```diff
@@ -55,8 +57,28 @@ export class SampleComponent {
   private readonly modalTemplate = viewChild<TemplateRef<any>>('modalTemplate');
   private sanitizer = inject(DomSanitizer);
   private readonly toastService = inject(SiToastNotificationService);
+  private translate = injectSiTranslateService();
 
-  protected markdownRenderer = getMarkdownRenderer(this.sanitizer);
+  protected markdownRenderer = getMarkdownRenderer(this.sanitizer, {
+    copyCodeButton: 'SI_MARKDOWN_RENDERER.COPY',
+    downloadTableButton: 'SI_MARKDOWN_RENDERER.DOWNLOAD',
```
[the remark this answers is not in the document]
Comment (Killusions) [reply src/app/examples/si-chat-messages/si-chat-container.ts]: Same here

File: src/app/examples/si-chat-messages/si-user-message.ts (hunk 4 above)
Comment (Killusions) [reply src/app/examples/si-chat-messages/si-user-message.ts]: And here
